### DISCUSSION TITLE
Menu: dropdown shifts too early when near the viewport edge?

### DISCUSSION THREAD

**Discussion by mmoo** [ariakit/ariakit#795] [Questions] [answered]
Hi Community. I'm not sure if it's a bug or I'm missing something. Please have a look at this example: https://stackblitz.com/edit/react-nqytpd
`placement` is set to `'bottom-start'` but the menu dropdown is shown on the right. If I change body margin to `8px` (or more) instead of `7px` it starts sticking to the left side of the button as expected. So there seems to be some gap between the viewport edge and the popper boundary? But I can't figure out how to control this. I need my menu just next to the viewport edge.
Please help.

**Comment by diegohaz** [comment, marked as answer]
Right now the value of `8` is hardcoded in the library and there's no way to change that (unless you come up with your own popper state). But you can disable this feature by passing `unstable_flip: false` to `useMenuState` (or any other state hook that uses popover state): https://stackblitz.com/edit/react-hswasg

Since this is experimental, just make sure to read this: https://reakit.io/docs/experimental/

We're aware of this limitation and currently discussing ways to make the popover configuration more flexible. See #606.

**Reply by mmoo** [reply]
Thank you!
